src/androidTest/java/medic/gateway/alert/SettingsDialogActivityTest.java

```diff
@@ -71,6 +63,7 @@ public void generic_shouldDisplayCancelButtonIfSettingsExist() throws Exception
 
 		// given
 		settingsStore().save(new Settings(http.url(), true, false, false));
+		Thread.sleep(2000);
```

Comment:
There is that but inside the tests, like [SettingsDialogActivityTest.java#L96](https://github.com/medic/cht-android/blob/49a3c206741b412d491e2b7f41780e0657c2d4ac/src/androidTestUnbrandedWebviewDebug/java/org/medicmobile/webapp/mobile/SettingsDialogActivityTest.java#L96) (the `withNoTimeout()` function allows to wait until the last action is performed), but this is in the "before" method that is executed in between, that for some reason if there is no gap of time in between, may cause the tests to fail because the test framework use a thread from the Android system to run the tests, and in new versions of this test framework the access to that thread is forbidden.

Reply:
Actually the comment above applies to the other reference of "Thread.sleep(2000);" you mention, not here that is inside a method. I just pushed a commit removing this call, that was my first attempt when refactoring the code.

Reply:
Ok it makes sense. I don't know of any better way and you clearly made your research, thanks for explaining it to me :raised_hands: